### PR TITLE
Option to ingest Kafka String columns in Avro Utf8 format

### PR DESCRIPTION
**Pull request by devonpallison**

### CODE REVIEW

File: .github/workflows/cla.yml
Comment (devinrsmith) [inline .github/workflows/cla.yml:1]: you should not be modifying this file.

File: extensions/kafka/src/main/java/io/deephaven/kafka/AvroImpl.java [hunk 1]
```diff
@@ -68,6 +69,8 @@ static final class AvroConsume extends Consume.KeyOrValueSpec {
         /** fields mapped to null are skipped. */
         private final Function<String, String> fieldPathToColumnName;
 
+        boolean useUTF8Strings = false;
```
Comment (devinrsmith) [inline extensions/kafka/src/main/java/io/deephaven/kafka/AvroImpl.java:72]: should be private, final.

**Comment by devonpallison** [comment]
I have read the CLA Document and I hereby sign the CLA

File: extensions/kafka/src/main/java/io/deephaven/kafka/AvroImpl.java [hunk 2]
```diff
@@ -453,7 +474,11 @@ private static void pushColumnTypesFromAvroField(
                 break;
             case ENUM:
             case STRING:
-                columnsOut.add(ColumnDefinition.ofString(mappedNameForColumn));
+                if (useUTF8Strings) {
+                    columnsOut.add(ColumnDefinition.of(mappedNameForColumn, Type.find(CharSequence.class)));
```
Comment (devinrsmith) [inline extensions/kafka/src/main/java/io/deephaven/kafka/AvroImpl.java:478]: We should probably use the more specific type, `org.apache.avro.util.Utf8`. Also, can store the type as a constant.

File: extensions/kafka/src/main/java/io/deephaven/kafka/AvroImpl.java [hunk 3]
```diff
@@ -84,6 +87,17 @@ static final class AvroConsume extends Consume.KeyOrValueSpec {
             this.fieldPathToColumnName = fieldPathToColumnName;
         }
 
+        private AvroConsume(final String schemaName,
```
Comment (devinrsmith) [inline extensions/kafka/src/main/java/io/deephaven/kafka/AvroImpl.java:90]: The other constructor can delegate to this one.

File: extensions/kafka/src/main/java/io/deephaven/kafka/AvroImpl.java (hunk 2 above)
Comment (devinrsmith) [reply extensions/kafka/src/main/java/io/deephaven/kafka/AvroImpl.java:478]: We'll also want to be very careful that the avro deserializer doesn't pool / keep hold of the `Utf8` objects; the javadoc explicitly states it's mutable:

```java
/**
 * A Utf8 string. Unlike {@link String}, instances are mutable. This is more
 * efficient than {@link String} when reading or writing a sequence of values,
 * as a single instance may be reused.
 */
```

**Review by devinrsmith — CHANGES_REQUESTED** [review]
In favor of this change, seems good. Couple of finishing points need to be addressed first.

**Comment by devinrsmith** [comment]
recheck

**Review by devinrsmith — COMMENTED** [review]
Does this provide DHE w/ enough hooks to actually use utf8?

I see `KafkaTools.avroSchemaToColumnDefinitions`, but no tie-ins w/  `KafkaTools.avroSpec`. Even though you've added `io.deephaven.kafka.AvroImpl.AvroConsume#useUTF8Strings`, that's a package private class, not publicly visible.

File: extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java [hunk 4]
```diff
@@ -483,6 +493,95 @@ public static KeyOrValueSpec avroSpec(final String schemaName) {
...
+         * @param fieldNameToColumnName A mapping specifying which Avro fields to include and what column name to use
+         *        for them; fields mapped to null are excluded.
+         * @return A spec corresponding to the schema provided.
+         */
+        @SuppressWarnings("unused")
+        public static KeyOrValueSpec avroSpecWithUTF8Strings(final String schemaName,
+                final String schemaVersion,
+                final Function<String, String> fieldNameToColumnName) {
+            return new AvroConsume(schemaName, schemaVersion, fieldNameToColumnName, true);
+        }
```
Comment (devinrsmith) [inline extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java:538]: Not a big fan of the proliferation of `WithUTF8Strings` messages; arguably, we may want to follow more of a builder pattern for cases like this in the future.

What about creating only having a single new method `avroSpec(String, String, Function, boolean)` that mirrors the new AvroConsume constructor?

Comment (devonpallison) [reply extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java:538]: Sure. Should be sufficient.

**Review by devinrsmith — APPROVED** [review]

**Pull request merged**